framework/include/materials/MaterialData.h

```diff
@@ -129,6 +129,15 @@ class MaterialData
    */
   const MaterialPropertyStorage & getMaterialPropertyStorage() const { return _storage; }
 
+  /**
+   * Wrapper for MaterialStorage::getPropertyId. Allows classes with a MaterialData object
+   * (i.e. MaterialPropertyInterface) to access material property IDs.
+   * @param prop_name The name of the material property
+   *
+   * @return An unsigned int corresponding to the property ID of the passed in prop_name
+   */
+  unsigned int getPropertyId(const std::string & prop_name) const { return _storage.getPropertyId(prop_name); }
```

Comment:
I would rather have this function.  It's logical to be able to ask this class for the ID... and it's inconsequential how the material property IDs are being held internally.